qiime2/metadata.py

```diff
@@ -91,6 +102,69 @@ def load(cls, path):
             raise ValueError(invalid_metadata_template % msg)
         return cls(df)
 
+    def merge(self, *others):
+        """Merge this ``Metadata`` object with other ``Metadata`` objects.
+
+        Returns a new ``Metadata`` object containing the merged contents of
+        this ``Metadata`` object and `others`. The merge is not in-place and
+        will always return a **new** merged ``Metadata`` object.
+
+        The merge will include only those IDs that are shared across **all**
+        ``Metadata`` objects being merged (i.e. the merge is an *inner join*).
+
+        Each column being merged must be unique; merging metadata with
```

Comment:
Good idea, I updated the docstring and error message text (see second "squash" commit)